browse/domain/metadata.py

```diff
@@ -77,6 +77,8 @@ class DocMetadata:
     submitter: Submitter
     """Submitter of the article."""
 
+    source_type: Optional[str] = None
```

Comment:
Please do something like this in the pacakge:
`source_type = Literal["tex", "ps", "html" .etc ]`
and then 
`source_type: Optional[source_type] = None`
Possibly a better name for the literal type than source_type?